puppet/lib/puppet/indirector/catalog/puppetdb.rb

```diff
@@ -291,7 +291,7 @@ def synthesize_edges(hash, catalog)
                 # case problem here: http://projects.puppetlabs.com/issues/19474
                 # Once that problem is solved and older versions of Puppet that have
                 # the bug are no longer supported we can probably remove this code.
-                unless other_ref =~ /^[A-Z][a-z0-9_-]*(::[A-Z][a-z0-9_-]*)*\[.*\]/m
+                unless other_ref =~ /^[A-Z][a-z0-9_]*(::[A-Z][a-z0-9_]*)*\[.*\]/m
```

Comment:
Yeah I looked into this a little bit, the regex's have changed a lot and we would essentially need to combine to regex's in some kind of sane way.